text/0000-Bundle-Load-API.md

```diff
@@ -0,0 +1,93 @@
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: (in a subsequent commit to the PR, fill me with the PR's URL)
+- CppMicroServices Issue: https://github.com/CppMicroServices/CppMicroServices/issues/376
+
+# Add Load API in Bundle class 
+
+## Summary
+
+Currently there is no API function in Bundle class to load bundles/dynamic libraries at runtime.
+This document discusses the need for such API , its possible design followed by solution.
+It should be analogues to Bundle::loadClass in the OSGi spec
+https://osgi.org/specification/osgi.core/7.0.0/framework.api.html#org.osgi.framework.Bundle.loadClass-String-
+
+## Motivation
+
+Clients of CppMicroServices such as declarative services rely on calling OS specific dlopen/LoadLibraryW calls for loading bundles/shared objects.
```

Comment:
I think there is an important distinction to be made here. Not all clients rely on loading libraries. Loading libraries is not something most clients should be doing. Generally speaking clients should be loading libraries via the bundle lifecycle (install, start, stop, uninstall). 
The capability to load libraries is required for specific use cases. One of these use cases is Declarative Services, which uses [the extender pattern](https://enroute.osgi.org/FAQ/400-patterns.html#extender-pattern) to provide some unique feature while leveraging existing common/shared features. I think making a point that this ```LoadBundle``` API is an extension point to be used for a specific reason is important. The specific use cases should then be called out in the "how we teach this" section. Users need to know when to (and not to) use the LoadBundle API.